### PR TITLE
:zap: Add support for Chronos-Bolt  models

### PR DESCRIPTION
**Pull request by abdulfatir**
*Issue #, if available:* N/A

*Description of changes:* This PR adds support for Chronos-Bolt models.

TODOs:

- [x] Update evaluation script
- [x] Fix and add tests for Bolt
- [x] Update docstrings
- [x] Update README example and mention Chronos-Bolt
- [x] Update results bar plot in README
- [x] Add versions for libraries in `pyproject.toml`
- [x] Check that the training and eval scripts work
- [x] Change `autogluon` -> `amazon` in model names

Post Merge:
- [ ] Update Citation style in README, both Github and HuggingFace repos
- [ ] Remove note about AutoGluon
- [ ] Update READMEs of original Chronos models to refer to Chronos-Bolt

NOTE: To be merged after Chronos-Bolt models are available under the `amazon` namespace on HF.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+    "accelerate~=0.32",
 ]
 
 [project.optional-dependencies]
 test = ["pytest~=8.0", "numpy~=1.21"]
 typecheck = ["mypy~=1.9"]
-training = ["gluonts[pro]", "numpy", "tensorboard", "typer", "typer-config", "joblib", "scikit-learn"]
-evaluation = ["gluonts[pro]", "datasets", "numpy", "typer"]
+training = ["gluonts[pro]~=0.15", "numpy~=1.21", "datasets~=2.18", "typer", "typer-config", "joblib", "scikit-learn", "tensorboard"]
+evaluation = ["gluonts[pro]~=0.15", "numpy~=1.21", "datasets~=2.18", "typer"]
```
Comment (lostella) [inline pyproject.toml:16]: We could also use numpy 2 with gluonts 0.16. Changes compared to 0.15 were rather minimal https://github.com/awslabs/gluonts/releases/tag/v0.16.0. But we can also do this in two steps and have a follow-up issue/PR

File: scripts/evaluation/evaluate.py [hunk 2]
```diff
@@ -12,10 +12,15 @@
...
 from gluonts.model.evaluation import evaluate_forecasts
-from gluonts.model.forecast import SampleForecast
+from gluonts.model.forecast import QuantileForecast, SampleForecast
 from tqdm.auto import tqdm
 
-from chronos import ChronosPipeline
+from chronos import (
+    BaseChronosPipeline,
+    ChronosBoltPipeline,
+    ChronosPipeline,
```
Comment (lostella) [inline scripts/evaluation/evaluate.py:21]: Are these needed now?

Comment (lostella) [reply scripts/evaluation/evaluate.py:21]: Yes they are

File: src/chronos/chronos.py [hunk 3]
```diff
@@ -538,6 +536,52 @@ def predict(
...
+
+        Returns
+        -------
+        quantiles
+            Tensor containing quantile forecasts.
+            Shape: (batch_size, prediction_length, num_quantiles)
+        mean
+            Tensor containing mean (point) forecasts.
+            Shape: (batch_size, prediction_length)
+        """
```
Comment (lostella) [inline src/chronos/chronos.py:570]: I see there are a lot of repeated docstrings (not just this class/method) should we find a way to make it less redundant? Not a lot of maintenance going on, but I'm wondering whether referencing the docstring of the base class and focus on the differences (added args) would be better

File: src/chronos/base.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (canerturkmen) [inline src/chronos/base.py:3]: should we give "links to original source code" to AG in case there is drift?

File: src/chronos/chronos.py (hunk 3 above)
Comment (abdulfatir) [reply src/chronos/chronos.py]: Added reference to the base class.

File: pyproject.toml (hunk 1 above)
Comment (abdulfatir) [reply pyproject.toml:16]: Let's do this in a follow-up then?

File: src/chronos/chronos_bolt.py [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+        prediction_length: Optional[int] = None,
+        limit_prediction_length: bool = False,
+    ) -> torch.Tensor:
+        """
+        Get forecasts for the given time series.
+
+        Refer to the base method (``BaseChronosPipeline.predict``)
+        for details on shared parameters.
+
+        limit_prediction_length
```
Comment (lostella) [inline src/chronos/chronos_bolt.py:437]: Missing header for additional parameters

File: src/chronos/chronos.py [hunk 6]
```diff
@@ -445,29 +443,24 @@ def embed(
...
-            is batch. In the latter case, use left-padding with
-            ``torch.nan`` to align series of different lengths.
-        prediction_length
-            Time steps to predict. Defaults to what specified
-            in ``self.model.config``.
+        Refer to the base method (``BaseChronosPipeline.predict``)
+        for details on shared parameters.
+
+        Additional Parameters:
+        ----------------------
```
Comment (lostella) [inline src/chronos/chronos.py:463]:
```suggestion
        Additional parameters
        ---------------------
```

File: src/chronos/chronos_bolt.py (hunk 5 above)
Comment (lostella) [reply src/chronos/chronos_bolt.py:437]:
```suggestion
        Additional parameters
        ---------------------
        limit_prediction_length
```

File: README.md [hunk 7]
```diff
@@ -17,7 +17,8 @@
 
 ## 🚀 News
 
-- **27 June 2024**: 🚀 [Released datasets](https://huggingface.co/datasets/autogluon/chronos_datasets) used in the paper and an [evaluation script](./scripts/README.md#evaluating-chronos-models) to compute the WQL and MASE scores reported in the paper. 
+- **25 Nov 2024**: ⚡️ Chronos-Bolt models released [on HuggingFace](https://huggingface.co/collections/amazon/chronos-models-65f1791d630a8d57cb718444). Chronos-Bolt models are more accurate (5% lower error), up to 250x faster and 20x more memory efficient than the original Chronos models of the same size!
```
Comment (lostella) [inline README.md:20]:
```suggestion
- **26 Nov 2024**: ⚡️ Chronos-Bolt models released [on HuggingFace](https://huggingface.co/collections/amazon/chronos-models-65f1791d630a8d57cb718444). Chronos-Bolt models are more accurate (5% lower error), up to 250x faster and 20x more memory efficient than the original Chronos models of the same size! 
```

**Review by lostella — APPROVED** [review]

**Pull request merged**